pkg/networkmanager/test-utils.js

```diff
@@ -0,0 +1,275 @@
+        assert.strictEqual(utils.ip4_from_text(checks[i][0]), checks[i][1]);
+    }
+});
+
+QUnit.test("ip4_from_text invalids", function() {
+    var checks = [
+        "",
+        "0",
+        "0.0",
+        "0.0.0",
+        "0.0.0.0.0",
+        "-1.2.3.4",
+        "foo",
+        "1.foo.3.4",
+        "1foo.2.3.4",
+        "1.2.3.400",
+        "1,2,3,4"
+    ];
+
+    assert_throws(utils.ip4_from_text, checks);
+});
+
+QUnit.test("ip4_to_text zero", function() {
+    utils.set_byteorder("be");
+    assert.strictEqual(utils.ip4_to_text(0, true), "");
+});
+
+QUnit.test("ip4_from_text empty", function() {
+    utils.set_byteorder("be");
+    assert.strictEqual(utils.ip4_from_text("", true), 0);
+});
+
+QUnit.test("ip4_prefix_from_text", function() {
+    var i, text;
+
+    var checks = [
+        "0.0.0.0",
+
+        "128.0.0.0",
+        "192.0.0.0",
+        "224.0.0.0",
+        "240.0.0.0",
+        "248.0.0.0",
+        "252.0.0.0",
+        "254.0.0.0",
+        "255.0.0.0",
+
+        "255.128.0.0",
+        "255.192.0.0",
+        "255.224.0.0",
+        "255.240.0.0",
+        "255.248.0.0",
+        "255.252.0.0",
+        "255.254.0.0",
+        "255.255.0.0",
+
+        "255.255.128.0",
+        "255.255.192.0",
+        "255.255.224.0",
+        "255.255.240.0",
+        "255.255.248.0",
+        "255.255.252.0",
+        "255.255.254.0",
+        "255.255.255.0",
+
+        "255.255.255.128",
+        "255.255.255.192",
+        "255.255.255.224",
+        "255.255.255.240",
+        "255.255.255.248",
+        "255.255.255.252",
+        "255.255.255.254",
+        "255.255.255.255"
+    ];
+
+    assert.expect(checks.length);
+
+    for (i = 0; i < checks.length; i++) {
+        assert.strictEqual(utils.ip4_prefix_from_text(checks[i]), i);
+    }
+});
+
+QUnit.test("ip4_prefix_from_text invalids", function() {
+    var checks = [
+        "",
+        "-1",
+        "foo",
+        "1foo",
+        "1.5",
+
+        "0.0",
+        "0.0.0",
+        "0.0.0.0.0",
+        "1.2.3.4",
+        "255.255.255.8",
+        "255.192.0.10"
+    ];
+
+    assert_throws(utils.ip4_prefix_from_text, checks);
+});
+
+QUnit.test("ip6_to/from_text", function() {
+    var i;
+
+    var checks = [
+        [ [ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 ],
+          "0:0:0:0:0:0:0:0"
+        ],
+        [ [ 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F ],
+          "1:203:405:607:809:a0b:c0d:e0f"
+        ],
+    ];
+
+    assert.expect(2*checks.length);
+
+    for (i = 0; i < checks.length; i++) {
+        assert.strictEqual(utils.ip6_to_text(cockpit.base64_encode(checks[i][0])), checks[i][1]);
+        assert.deepEqual(cockpit.base64_decode(utils.ip6_from_text(checks[i][1])), checks[i][0]);
+    }
+});
+
+QUnit.test("ip6_from_text shortcuts", function() {
```

Comment:
the actual code uses trim to ignore whitespace (e.g. [here](https://github.com/cockpit-project/cockpit/pull/5598/files#diff-27d9c791d3b36baa7fa523a84d2ca62eR200))
I recommend adding a few tests with extra whitespace in the individual parts so we know that the trimming works properly.